mbed_host_tests/host_tests_plugins/module_copy_mps2.py

```diff
@@ -17,96 +17,91 @@
-        """ This function will alter image cfg file
-
-        @details Main goal of this function is to change number of images to 1, comment all
-                 existing image entries and append at the end of file new entry with test path
-
-        @return True when all steps were successful
-        """
-        MBED_SDK_TEST_STAMP = 'test suite entry'
-        image_path = join(disk, images_cfg_path, image_name)
-        new_file_lines = [] # New configuration file lines (entries)
-
-        # Check each line of the image configuration file
-        try:
-            with open(image_path, 'r') as file:
-                for line in file:
-                    if re.search('^TOTALIMAGES', line):
-                        # Check number of total images, should be 1
-                        new_file_lines.append(re.sub('^TOTALIMAGES:[\t ]*[\d]+', 'TOTALIMAGES: 1', line))
-                    elif re.search('; - %s[\n\r]*$'% MBED_SDK_TEST_STAMP, line):
-                        # Look for test suite entries and remove them
-                        pass    # Omit all test suite entries
-                    elif re.search('^IMAGE[\d]+FILE', line):
-                        # Check all image entries and mark the ';'
-                        new_file_lines.append(';' + line)   # Comment non test suite lines
-                    else:
-                        # Append line to new file
-                        new_file_lines.append(line)
-        except IOError as e:
-            return False
-
-        # Add new image entry with proper commented stamp
-        new_file_lines.append('IMAGE0FILE: %s    ; - %s\r\n'% (image0file_path, MBED_SDK_TEST_STAMP))
-
-        # Write all lines to file
-        try:
-            with open(image_path, 'w') as file:
-                for line in new_file_lines:
-                    file.write(line),
-        except IOError:
-            return False
+    def generic_mbed_copy(self, image_path, destination_disk):
+        """! Generic mbed copy method for "mbed enabled" devices.
 
-        return True
+        @param image_path Path to binary file to be flashed
+        @param destination_disk Path to destination (mbed mount point)
 
-    def mps2_select_core(self, disk, mobo_config_name=""):
-        """ Function selects actual core
-        """
-        # TODO: implement core selection
-        pass
+        @details It uses standard python shutil function to copy image_file (target specific binary) to device's disk.
 
-    def mps2_switch_usb_auto_mounting_after_restart(self, disk, usb_config_name=""):
-        """ Function alters configuration to allow USB MSD to be mounted after restarts
+        @return Returns True if copy (flashing) was successful
         """
-        # TODO: implement USB MSD restart detection
-        pass
+        result = True
+        destination_path = os.path.join(destination_disk,"mbed.bin")
+        try:
+            copy(image_path, destination_path)
```

Comment:
The real question is: is there magic sauce in the windows copy command that shutil.copy does not do?